From ee84e10f3daba3050b17904c34b05db0765736d4 Mon Sep 17 00:00:00 2001
From: Viktor Mihajlovski <mihajlov@linux.vnet.ibm.com>
Date: Wed, 11 Sep 2013 16:45:35 +0200
Subject: [PATCH 15/60] VSMS: Set resource types for default devices

The default graphics and input devices were built without their
resource types being set correctly. This has not hurted yet.
Future changes will however require that the device resource
type is matching the actual device type.

Signed-off-by: Viktor Mihajlovski <mihajlov@linux.vnet.ibm.com>
Reviewed-by: Boris Fiuczynski <fiuczy@linux.vnet.ibm.com>
Signed-off-by: John Ferlan <jferlan@redhat.com>
---
 src/Virt_VirtualSystemManagementService.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Virt_VirtualSystemManagementService.c b/src/Virt_VirtualSystemManagementService.c
index 79dec73..6629b35 100644
--- a/src/Virt_VirtualSystemManagementService.c
+++ b/src/Virt_VirtualSystemManagementService.c
@@ -581,6 +581,7 @@ static bool default_graphics_device(struct domain *domain)
                 return false;
         }
 
+        domain->dev_graphics->type = CIM_RES_TYPE_GRAPHICS;
         domain->dev_graphics->dev.graphics.type = strdup("vnc");
         domain->dev_graphics->dev.graphics.dev.vnc.port = strdup("-1");
         domain->dev_graphics->dev.graphics.dev.vnc.host = strdup("127.0.0.1");
@@ -609,6 +610,7 @@ static bool default_input_device(struct domain *domain)
                 return false;
         }
 
+        domain->dev_input->type = CIM_RES_TYPE_INPUT;
         domain->dev_input->dev.input.type = strdup("mouse");
 
         if (domain->type == DOMAIN_XENPV) {
-- 
2.1.0

